firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/persistence/CrashlyticsReportPersistence.java

```diff
@@ -192,17 +192,20 @@ public void finalizeReports(String currentSessionId) {
 
         CrashlyticsReport report =
             TRANSFORM.reportFromJson(readTextFile(new File(sessionDirectory, REPORT_FILE_NAME)));
-        final String sessionId = report.getSession().getIdentifier();
 
-        if (userId != null) {
-          report = report.withUserId(userId);
-        }
+        // TODO: Handle null case
+
+        if (report != null) {
+          report = report.withSessionEndFields(sessionEndTime, isHighPriorityReport, userId);
```

Comment:
this finalizeReports call is pretty nested. Suggestion:

Refactor to make it so this method is much shorter, like:
```
writeReport(...);
recursiveDelete(sessionDirectory);
```
Then in writeReport(...), flip the two nested ifs into early terminating statements:

```
if (eventFiles.isEmpty()) {
  return;
}
...
if (report == null) {
  // TODO: Handle null case
  return
}```

Reply:
👍

Reply:
nice!!